sourced/ml/core/algorithms/token_parser.py

```diff
@@ -49,6 +52,14 @@ def __init__(self, stem_threshold=STEM_THRESHOLD, max_token_length=MAX_TOKEN_LEN
         self._single_shot = single_shot
         self._save_token_style = save_token_style
         self._attach_upper = attach_upper
+        if nn_model == "default":
+            self._nn_model = self.NN_MODEL
+            self._init_nn()
+        elif nn_model != None:
+            self._nn_model = nn_model
+            self._init_nn()
+        else:
+            self._nn_model = None
```

Comment:
```suggestion
self._nn_model = nn_model
self._init_nn()
```

Disregarding any `if`-s - they should be removed.